sdk/bulk.go

```diff
@@ -82,9 +95,49 @@ func (s SDK) BulkDecrypt(ctx context.Context, opts ...BulkDecryptOption) error {
 	tdfDecryptors := make(map[string]decryptor)
 	policyTDF := make(map[string]*BulkTDF)
 
+	switch bulkReq.TDFType {
+	case Nano:
+		dummy := &NanoTDFReaderConfig{}
+		for _, opt := range bulkReq.NanoTDFDecryptOptions {
+			_ = opt(dummy)
+		}
+
+		if !dummy.ignoreAllowList && dummy.kasAllowlist == nil {
+			// if no kasAllowlist is set, we get the allowlist from the registry
+			allowlist, err := allowListFromKASRegistry(ctx, s.KeyAccessServerRegistry, s.conn.Target())
+			if err != nil {
+				return fmt.Errorf("failed to get allowlist from registry: %w", err)
+			}
+			bulkReq.NanoTDFDecryptOptions = append(
+				bulkReq.NanoTDFDecryptOptions,
+				withNanoKasAllowlist(allowlist),
+			)
+		}
+	case Standard:
+		dummy := &TDFReaderConfig{}
+		for _, opt := range bulkReq.TDF3DecryptOptions {
+			_ = opt(dummy)
+		}
+		if !dummy.ignoreAllowList && dummy.kasAllowlist == nil {
+			// if no kasAllowlist is set, we get the allowlist from the registry
+			if !dummy.ignoreAllowList && dummy.kasAllowlist == nil {
+				// if no kasAllowlist is set, we get the allowlist from the registry
+				allowlist, err := allowListFromKASRegistry(ctx, s.KeyAccessServerRegistry, s.conn.Target())
+				if err != nil {
+					return fmt.Errorf("failed to get allowlist from registry: %w", err)
+				}
+				bulkReq.TDF3DecryptOptions = append(
+					bulkReq.TDF3DecryptOptions,
+					withKasAllowlist(allowlist),
+				)
+			}
```

Comment:
Seems duplicative? 

Also, instead of building a dummy, you could have a second phase for the options, like defaults, bulk defaults (what is the dummy here), then the actual stuff.

But yeah this is good enough and we can optimize later, esp. when if we (hopefully) get the 'unified' API out like we did for web-sdk